python/packages/autogen-ext/src/autogen_ext/tools/azure/_ai_search.py

```diff
@@ -0,0 +1,1133 @@
+
+            # Perform a semantic search
+            try:
+                result = await semantic_search_tool.run_json(args={"query": "latest advances in neural networks"})
+                print(f"Found {len(result.results)} results")
+            except Exception as e:
+                print(f"Search error: {e}")
+
+        .. code-block:: python
+
+            # Vector search example
+            # Create vector search tool
+            vector_search_tool = AzureAISearchTool.load_component(
+                ComponentModel(
+                    provider="autogen_ext.tools.azure.AzureAISearchTool",
+                    config={
+                        "name": "VectorSearch",
+                        "description": "Vector search with Azure AI Search",
+                        "endpoint": "https://your-search-service.search.windows.net",
+                        "index_name": "your-index",
+                        "api_version": "2023-10-01-Preview",
+                        "credential": {"api_key": "your-api-key"},
+                        "query_type": "vector",
+                        "vector_fields": ["embedding"],
+                        "select_fields": ["id", "content", "title", "source"],
+                        "openai_client": openai_client,
+                        "embedding_model": "text-embedding-ada-002",
+                        "top": 5,
+                    },
+                )
+            )
+
+            # Perform a vector search with a text query (will be converted to vector)
+            result = await vector_search_tool.run_json(args={"query": "quantum computing algorithms"})
+
+            # Or use a pre-computed vector directly
+            vector = [0.1, 0.2, 0.3, 0.4]  # Example vector (would be much longer in practice)
+            result = await vector_search_tool.run_json(args={"vector": vector})
+
+    Using with AutoGen Agents:
+        .. code-block:: python
+
+            # Set up the search tool with an agent
+            from autogen import ConversableAgent
+            from autogen_core import ComponentModel
+            from autogen_ext.tools.azure import AzureAISearchTool
+
+            # Create the search tool
+            search_tool = AzureAISearchTool.load_component(
+                ComponentModel(
+                    provider="autogen_ext.tools.azure.AzureAISearchTool",
+                    config={
+                        "name": "DocumentSearch",
+                        "endpoint": "https://your-search-service.search.windows.net",
+                        "index_name": "your-index",
+                        "credential": {"api_key": "your-api-key"},
+                        "query_type": "semantic",
+                        "openai_client": openai_client,
+                        "embedding_model": "text-embedding-ada-002",
+                    },
+                )
+            )
+
+            # Create an assistant with the search tool
+            assistant = ConversableAgent(
```

Comment:
This example is based on v0.2 API. Let's use the latest API. See the quick start example in AgentChat. 

Also, I think this example should be moved to `OpenAIAzureAISearchTool`.